modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/Enrich.scala

```diff
@@ -69,16 +71,21 @@ object Enrich {
   def run[F[_]: Concurrent: ContextShift: Clock](env: Environment[F]): Stream[F, Unit] = {
     val registry: F[EnrichmentRegistry[F]] = env.enrichments.get.map(_.registry)
     val enrich: Enrich[F] = enrichWith[F](registry, env.blocker, env.igluClient, env.sentry, env.metrics.enrichLatency)
-    val badSink: BadSink[F] = _.evalTap(_ => env.metrics.badCount).through(env.bad)
-    val goodSink: GoodSink[F] = _.evalTap(_ => env.metrics.goodCount).through(env.good)
 
     env.source
       .pauseWhen(env.pauseEnrich)
       .evalTap(_ => env.metrics.rawCount)
       .parEvalMapUnordered(ConcurrencyLevel)(enrich)
       .flatMap(_.decompose[BadRow, EnrichedEvent])
-      .observeEither(badSink, goodSink)
-      .void
+      .parEvalMapUnordered(ConcurrencyLevel)(Payload.sinkEither(badSink(env), goodSink(env)))
```

Comment:
TODO: I think this should not be parallelised.  Payloads should sunk in order, so the final ack means all payloads of a batch have been sunk.

Reply:
> so the final ack means all payloads of a batch have been sunk.

This would be true with Kinesis where we ack on offset, and not a message. I wonder if this is also true for PubSub.

We already have `parEvalMapUnordered` before, so I think that we already lost the order of messages at this point.

Reply:
My comment refers to PubSub too, where we ack every message.  This is specifically for messages that comprise multiple events.

Line 79 decomposes a each message into stream of multiple events:
```scala
.flatMap(_.decompose[BadRow, EnrichedEvent])
```
The `decompose` method sets things up so that the final event in the batch (for a single pubsub message) will ack the message.  But this only works if we process the batch of events in order.

Reply:
I get it, thanks !